pkg/models/repository_test.go

```diff
@@ -34,4 +35,43 @@ func (s *RepositorySuite) TestRepositoriesCreate() {
 	err := tx.Where("url = ?", repo.URL).First(&found).Error
 	assert.Nil(s.T(), err)
 	assert.NotEmpty(s.T(), found.UUID)
+	assert.True(s.T(), strings.HasSuffix(found.URL, "/")) // test trailing slash added during creation
+}
+
+func (s *ModelsSuite) TestCleanupURL() {
+	tx := s.tx
+	var found Repository
+
+	type TestCase struct {
+		given    string
+		expected string
+	}
+
+	testCases := []TestCase{
+		{
+			given:    "https://example1.com",
```

Comment:
can you use .test or .invalid?
```suggestion
			given:    "https://example1.test",
```
I see that domain has some text, `example.com` would also display some test.